pkg/ops/miningManager.go

```diff
@@ -152,102 +156,140 @@ func (mgr *MiningMgr) Start(ctx context.Context) {
+// with an error which can be ignored.
+func (mgr *MiningMgr) submit(solution *pow.Result, ctx context.Context) bool {
+	lastSubmit, err := mgr.lastMinerSubmit()
+	if err != nil {
+		level.Error(mgr.logger).Log("msg", "checking last submit time", "err", err)
+	} else if lastSubmit < mgr.cfg.MinSubmitPeriod {
+		level.Debug(mgr.logger).Log("msg", "min transaction submit threshold hasn't passed", "minSubmitPeriod", mgr.cfg.MinSubmitPeriod, "lastSubmit", lastSubmit)
+		return false
+	}
 
-		// Time to check for a new challenge.
-		case <-ticker.C:
-			mgr.newWork()
+	var ids []int64
+	for _, id := range solution.Work.Challenge.RequestIDs {
+		ids = append(ids, id.Int64())
+	}
+	level.Debug(mgr.logger).Log("msg", "submitting solution", "reqIDs", fmt.Sprintf("%+v", ids))
+
+	var profitPercent int64
+	var slot int64
+	if mgr.cfg.ProfitThreshold > 0 {
+		profitPercent, slot, err = mgr.profit()
+		if err != nil {
+			level.Error(mgr.logger).Log("msg", "submit solution profit check", "err", err)
+			return false
+		}
+		if profitPercent != -1 && profitPercent < int64(mgr.cfg.ProfitThreshold) {
+			level.Debug(mgr.logger).Log("msg", "transaction not profitable, so will wait for the next cycle")
+			return false
 		}
 	}
+
+	tx, err := mgr.solHandler.Submit(ctx, solution)
+	if err != nil {
+		level.Error(mgr.logger).Log("msg", "submiting a solution", "err", err)
+		mgr.submitFailCount.Inc()
+		return false
+	}
+	level.Debug(mgr.logger).Log("msg", "submited a solution", "txHash", tx.Hash().String())
+	mgr.saveGasUsed(ctx, tx)
+	mgr.submitCount.Inc()
+	mgr.submitProfit.With(prometheus.Labels{"slot": strconv.Itoa(int(slot))}).(prometheus.Gauge).Set(float64(profitPercent))
+
+	return true
 }
 
 // newWork is non blocking worker that sends new work to the pow workers
 // or re-sends a current pending solution to the submitter when the challenge hasn't changes.
 func (mgr *MiningMgr) newWork() {
-	go func() {
-		if mgr.cfg.EnablePoolWorker {
-			mgr.tasker.GetWork(mgr.toMineInput)
-		} else {
-			// instantSubmit means 15 mins have passed so
-			// the difficulty now is zero and any solution/nonce will work so
-			// can just submit without sending to the miner.
-			work, instantSubmit := mgr.tasker.GetWork(nil)
-			if instantSubmit {
-				mgr.solutionOutput <- &pow.Result{Work: work, Nonce: "anything will work"}
-			} else {
-				// It sends even nil work to indicate that no new challenge is available.
-				if work == nil {
-					mgr.solutionOutput <- nil
-					return
-				}
+	if mgr.cfg.EnablePoolWorker {
+		mgr.tasker.GetWork(mgr.toMineInput)
+	} else {
+		work := mgr.tasker.GetWork(nil)
+		if work == nil {
+			return
+		}
+		var ids []int64
+		for _, id := range work.Challenge.RequestIDs {
+			ids = append(ids, id.Int64())
+		}
+		// For oracle blocks that are more than 15mins old can submit any solution.
+		// No need to send it for minging.
+		if mgr.isInstantSubmit() {
```

Comment:
Now this only get's called when a challenge is new (it returns on line 231)...
You want it to check for instantSubmit every trackerCycle